signerv2/remote_signer_client.go

```diff
@@ -0,0 +1,107 @@
+package signerv2
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/Layr-Labs/eigensdk-go/utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+type RpcRequest struct {
+	JsonRPC string      `json:"jsonrpc"`
+	Method  string      `json:"method"`
+	Params  interface{} `json:"params"`
+	ID      int         `json:"id"`
+}
+
+type RemoteSignerClient interface {
+	SignTransaction(from common.Address, tx *types.Transaction) (*types.Transaction, error)
+}
+
+// RemoteSigner is a client for a remote signer
+// It implements Consensys Web3 Signer
+// Reference: https://docs.web3signer.consensys.io/reference/api/json-rpc
+type RemoteSigner struct {
+	url    string
+	client http.Client
+}
+
+func NewRemoteSignerClient(url string) RemoteSignerClient {
+	client := http.Client{}
+	return &RemoteSigner{client: client, url: url}
+}
+
+func (r RemoteSigner) SignTransaction(
+	from common.Address,
+	tx *types.Transaction,
+) (*types.Transaction, error) {
+	method := "eth_signTransaction"
+	id := 1
+	params := []map[string]string{
+		{
+			"from":     from.Hex(),
+			"to":       tx.To().Hex(),
+			"gas":      utils.Add0x(fmt.Sprintf("%x", tx.Gas())),
+			"gasPrice": utils.Add0x(hex.EncodeToString(tx.GasPrice().Bytes())),
+			"nonce":    utils.Add0x(fmt.Sprintf("%x", tx.Nonce())),
+			"data":     utils.Add0x(hex.EncodeToString(tx.Data())),
+		},
+	}
+
+	request := RpcRequest{
+		JsonRPC: "2.0",
+		Method:  method,
+		Params:  params,
+		ID:      id,
+	}
+
+	jsonData, err := json.Marshal(request)
+	if err != nil {
+		fmt.Println("Error marshalling request:", err)
+		return nil, err
```

Comment:
return a wrapped error instead of printing errors everywhere. I want to switch to this convention instead of cluttering logs with multi line stuff

Reply:
ahh good point. I should maybe include logger in overall signer too so we don't use `println`

Reply:
actually nevermind,. it's not client to log. I will return the wrapper error

Reply:
7a6df19465ed37185f3719c841bd4ec3d2e9ddaf